File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
    let order = ""
    let map = new Map()
    strs.forEach((value, index) => {
        let check_arr = value.split("")
        order = check_arr.sort().join("")
        if (map.has(order)) {
            map.get(order).push(value)
        }
        else {
            map.set(order, [value])
        }
    })
    let ans = []
    for (let [key, val] of map) {
        ans.push(val)
    }
    return ans
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result = []
        collector = {}

        for i in range(len(strs)):
            sorted_key =  ",".join((sorted(list(strs[i]))))
            if sorted_key not in collector:
                collector[sorted_key] = []

            collector[sorted_key].append(strs[i])
        
        for key,value in collector.items():
            result.append(value)
        return result